compiler/src/cobalt/Parser/ExprParser.hs

```diff
@@ -76,57 +90,37 @@ assignParser = do
             return (immutable, varNames, varType)
         return start
 
-aExpr :: Parser AExpr
-aExpr = makeExprParser aTerm aOperators
+aTerm :: Parser AExpr
+aTerm
+    =   parens aExpr
+    <|> Var      <$> identifier
+    <|> IntConst <$> integerParser
+    <|> IntConst <$> integerParser
+    <|> IntConst <$> integerParser
+    <|> IntConst <$> integerParser
 
 bExpr :: Parser BExpr
 bExpr = makeExprParser bTerm bOperators
 
-aOperators :: [[Operator Parser AExpr]]
-aOperators =
-    [ [Prefix (Neg <$ symbol "-") ]
-    , [ InfixL (ABinary Multiply <$ symbol "*")
-      , InfixL (ABinary Divide   <$ symbol "/") ]
-    , [ InfixL (ABinary Add      <$ symbol "+")
-      , InfixL (ABinary Subtract <$ symbol "-") ]
-    ]
-
 bOperators :: [[Operator Parser BExpr]]
 bOperators =
     [ [Prefix (Not <$ rword "not") ]
     , [InfixL (BBinary And <$ rword "and")
       , InfixL (BBinary Or <$ rword "or") ]
     ]
 
-aTerm :: Parser AExpr
-aTerm
-    =   parens aExpr
-    <|> Var      <$> identifier
-    <|> IntConst <$> integerParser
-    <|> IntConst <$> integerParser
-    <|> IntConst <$> integerParser
-    <|> IntConst <$> integerParser
-
 bTerm :: Parser BExpr
 bTerm =  parens bExpr
   <|> (BoolConst True  <$ rword "True")
   <|> (BoolConst False <$ rword "False")
   <|> rExpr
 
-rExpr :: Parser BExpr
-rExpr = do
-  a1 <- aExpr
-  op <- relation
-  a2 <- aExpr
-  return (RBinary op a1 a2)
-
-relation :: Parser RBinOp
-relation = (symbol ">" *> pure Greater)
-  <|> (symbol "<" *> pure Less)
-
 expressionParser :: Parser Expr
 expressionParser
-    = Block <$> many statementParser
+    = identifierParser
```

Comment:
This dramatically changes expected semantics. Is it just temporary?

Reply:
Originally an expression could be many statements (which doesn't make sense) and now it can only parse individual expressions. What could be the problem with this?

Reply:
It only parses `Identifier`, no any other expr like `Call`

Reply:
I see. After all the changes I haven't added any of the new ones. I'll write up a list of parsers we need to get tested etc as there are still some things missing like this since the change over to the new AST.